src/main/java/org/logstash/dissect/NormalField.java

```diff
@@ -0,0 +1,102 @@
+package org.logstash.dissect;
+
+import com.logstash.Event;
+
+import java.util.Map;
+
+class NormalField implements Field {
+    private int _ordinal;
+    protected final String _name;
+
+    protected Delim _join;
+    protected Delim _next;
+
+    private static final Field missing = new NormalField("missing field", 100000);
+
+    public static Field create(String s) {
+        return new NormalField(s);
+    }
+
+    public static Field getMissing() {
+        return missing;
+    }
+
+    public NormalField(String s) {
+        this(s, 1);
+    }
+
+    protected NormalField(String s, int ord) {
+        _name = s;
+        _ordinal = ord;
+    }
+
+    protected static String leftChop(String s) {
```

Comment:
I have moved it to the new FieldFactory and renamed it to `removeLeadingCharIfPresent`